google/cloud/opentelemetry/quickstart/README.md

```diff
@@ -0,0 +1,201 @@
+`opentelemetry-cpp`. That is v1.9.0, circa 2023-05.
+
+## Using with Bazel
+
+> :warning: If you are using Windows or macOS there are additional instructions
+> at the end of this document.
+
+1. Install Bazel using [the instructions][bazel-install] from the `bazel.build`
+   website.
+
+1. Compile this example using Bazel:
+
+   ```bash
+   cd $HOME/google-cloud-cpp/google/cloud/opentelemetry/quickstart
+   bazel build ...
+   ```
+
+   Note that Bazel automatically downloads and compiles all dependencies of the
+   project. As it is often the case with C++ libraries, compiling these
+   dependencies may take several minutes.
+
+1. Run the example, changing the placeholders to appropriate values:
+
+   ```bash
+   bazel run :quickstart -- [BUCKET_NAME] [PROJECT_ID]
+   ```
+
+### Details
+
+Note the following two feature flags explicitly set in the `.bazelrc`. Together,
+these flags enable OpenTelemetry tracing instrumentation in `google-cloud-cpp`.
+
+```sh
+# Required for OpenTelemetry + Abseil compatibility
+build --@io_opentelemetry_cpp//api:with_abseil
+
+# Enables tracing instrumentation in google-cloud-cpp
+build --@google_cloud_cpp//:enable-experimental-opentelemetry
+```
+
+Also note that we explicitly load OpenTelemetry's dependencies in the
+`WORKSPACE.bazel`.
+
+## Using with CMake
+
+> :warning: If you are using Windows or macOS there are additional instructions
+> at the end of this document.
+
+### Building from Source
+
+#### Building `google-cloud-cpp`
+
+The [packaging guide] contains instructions for how to build from source.
+
+The Google Cloud exporters are built as part of the feature,
+`experimental-opentelemetry`. This quickstart also uses the GCS client, from the
+`storage` feature.
+
+To enable these features, add the following to your CMake configuration command:
+
+```
+-DGOOGLE_CLOUD_CPP_ENABLE="storage,experimental-opentelemetry"
+```
+
+#### Fetching `opentelemetry-cpp`
+
+Here is an example command to fetch and install `opentelemetry-cpp` on Fedora
+Linux.
+
+```Dockerfile
+WORKDIR /var/tmp/build/
+RUN curl -sSL https://github.com/open-telemetry/opentelemetry-cpp/archive/v1.9.0.tar.gz | \
+    tar -xzf - --strip-components=1 && \
+    cmake \
+        -DCMAKE_CXX_STANDARD=14 \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE \
+        -DBUILD_SHARED_LIBS=ON \
+        -DWITH_EXAMPLES=OFF \
+        -DWITH_ABSEIL=ON \
+        -DBUILD_TESTING=OFF \
+        -DOPENTELEMETRY_INSTALL=ON \
+        -H. -Bcmake-out && \
+    cmake --build cmake-out --target install && \
+    ldconfig && cd /var/tmp && rm -fr build
+```
+
+#### Details
+
+We must supply the `-DWITH_ABSEIL=ON` flag when compiling `opentelemetry-cpp`
+for compatibility with Abseil.
+
+We must also ensure that the `-DCMAKE_CXX_STANDARD` used to compile
+`opentelemetry-cpp` matches the language standard used to compile `abseil-cpp`.
+It does not need to be C++14.
```

Comment:
Okay, but what does it need to be?  Maybe say "OpenTelemetry supports C++ >= 11, while `google-cloud-cpp` requires C++ >= 14. You can use `-DCMAKE_CXX_STANDARD=14`, `-DCMAKE_CXX_STANDARD=17`, or higher".